### PR TITLE
fix: Backport minor updates for Core 18 functionality

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

Updates that cover up errors with Core 18

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/evolution/SimplifiedMasternodeListManager.java [hunk 1]
```diff
@@ -255,6 +256,17 @@ public void processMasternodeListDiff(SimplifiedMasternodeListDiff mnlistdiff, b
...
+            if (!isQuorumRotationEnabled) {
+                quorumList.forEachQuorum(true, new SimplifiedQuorumList.ForeachQuorumCallback() {
+                    @Override
+                    public void processQuorum(FinalCommitment finalCommitment) {
+                        if (finalCommitment.getLlmqType() == params.getLlmqTypeInstantSendDIP24().getValue()) {
+                            isQuorumRotationEnabled = true;
+                        }
+                    }
+                });
+            }
```
Comment (HashEngineering) [inline core/src/main/java/org/bitcoinj/evolution/SimplifiedMasternodeListManager.java:268]: This code determines if DIP24 is active.  We look for quorums with DIP24 quorum types.

File: core/src/main/java/org/bitcoinj/quorums/InstantSendManager.java [hunk 2]
```diff
@@ -463,8 +463,10 @@ boolean processPendingInstantSendLocks() throws BlockStoreException
             }
         }
 
-        if(context.masternodeSync.hasVerifyFlag(MasternodeSync.VERIFY_FLAGS.BLS_SIGNATURES))
+        if(context.masternodeSync.hasVerifyFlag(MasternodeSync.VERIFY_FLAGS.BLS_SIGNATURES) &&
+                !context.masternodeListManager.isQuorumRotationEnabled()) {
             batchVerifier.verify();
+        }
```
Comment (HashEngineering) [inline core/src/main/java/org/bitcoinj/quorums/InstantSendManager.java:469]: We will only verify signatures if DIP24 is not active.  If it is active, we cannot verify signatures.

File: core/src/main/java/org/bitcoinj/core/BitcoinSerializer.java [hunk 3]
```diff
@@ -286,7 +286,9 @@ private Message makeMessage(String command, int length, byte[] payloadBytes, byt
         } else if(command.equals("qsendrecsigs")) {
             return new QuorumSendRecoveredSignatures(params);
         } else if(command.equals("islock")) {
-            return new InstantSendLock(params, payloadBytes);
+            return new InstantSendLock(params, payloadBytes, InstantSendLock.ISLOCK_VERSION);
+        } else if(command.equals("isdlock")) {
+            return new InstantSendLock(params, payloadBytes, InstantSendLock.ISDLOCK_VERSION);
```
Comment (HashEngineering) [inline core/src/main/java/org/bitcoinj/core/BitcoinSerializer.java:291]: this change allows us to receive both new and old InstantSend messages (`isdlock` is new)

File: core/src/main/java/org/bitcoinj/quorums/InstantSendManager.java (hunk 2 above)
Comment (HashEngineering) [reply core/src/main/java/org/bitcoinj/quorums/InstantSendManager.java:469]: Signatures that are without errors after this call are assumed to be verified.

File: core/src/main/java/org/bitcoinj/quorums/FinalCommitment.java [hunk 4]
```diff
@@ -55,6 +57,11 @@ protected void parse() throws ProtocolException {
 
         llmqType = readBytes(1)[0];
         quorumHash = readHash();
+        if (getVersion() >= INDEXED_QUORUM_VERSION) {
+            quorumIndex = readUint16();
+        } else {
+            quorumIndex = 0;
+        }
```
Comment (HashEngineering) [inline core/src/main/java/org/bitcoinj/quorums/FinalCommitment.java:64]: this change will read the new format of quorums in Core 18.  Without this, the masternode list synchronization will fail.

File: core/src/main/java/org/bitcoinj/core/NetworkParameters.java [hunk 5]
```diff
@@ -623,7 +623,8 @@ public static enum ProtocolVersion {
         BLOOM_FILTER_BIP111(MINIMUM.getBitcoinProtocolVersion()+1),
         @Deprecated
         DMN_LIST(70214),
-        CURRENT(70219);
+        ISDLOCK(70220),
+        CURRENT(70220);
```
Comment (HashEngineering) [inline core/src/main/java/org/bitcoinj/core/NetworkParameters.java:627]: 70220 is the protocol version for ISDLOCK (InstantSend Deterministic)

**Review by Syn-McJ — APPROVED** [review]
Looks great

**Pull request merged**